extensions/amp-render/0.1/amp-render.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+
+
+class AmpRender extends AMP.BaseElement {
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  isReadyToBuild() {
+    return this.element.firstChild != null;
+  }
+
+  /** @override */
+  buildCallback() {
+  }
+
+  /** @override */
+  layoutCallback() {
+    return AMP.renderTemplate(this.element, {name: 'Dima', time: new Date()}).
```

Comment:
A very limited `fetch` polyfill would be great. I believe the full spec might be way too complex.